fetch-nixpkgs.nix

```diff
@@ -1,10 +1,8 @@
 let
-  fetchNixpkgs = rev: sha256unpacked: builtins.fetchTarball {
-    url = "https://github.com/NixOS/nixpkgs/archive/${rev}.tar.gz";
-    sha256 = sha256unpacked;
-  };
-  nixpkgsJson = builtins.fromJSON (builtins.readFile ./nixpkgs-src.json);
-  pkgs_path = fetchNixpkgs nixpkgsJson.rev nixpkgsJson.sha256unpacked;
+  # temporary hack until scripts/nix-shell.sh ceases to use -p
+  pkgs_path = import ./fetchNixpkgs.nix (builtins.fromJSON (builtins.readFile ./nixpkgs-src.json));
```

Comment:
It looks like a lot of this is just copying what they're now doing in Cardano? I don't inherently object, but it's not clearly related to this PR, and I repeat my desire for a better way of sharing the code...